pkg/master/storageversionhashdata/data.go

```diff
@@ -0,0 +1,108 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package storageversionhashdata is for test only.
+package storageversionhashdata
+
+import (
+	"k8s.io/apimachinery/pkg/util/sets"
+)
+
+// NoStorageVersionHash lists resources that legitimately with empty storage
+// version hash.
+var NoStorageVersionHash = sets.NewString(
+	"v1/bindings",
+	"v1/componentstatuses",
+	"authentication.k8s.io/v1/tokenreviews",
+	"authorization.k8s.io/v1/localsubjectaccessreviews",
+	"authorization.k8s.io/v1/selfsubjectaccessreviews",
+	"authorization.k8s.io/v1/selfsubjectrulesreviews",
+	"authorization.k8s.io/v1/subjectaccessreviews",
+	"authentication.k8s.io/v1beta1/tokenreviews",
+	"authorization.k8s.io/v1beta1/localsubjectaccessreviews",
+	"authorization.k8s.io/v1beta1/selfsubjectaccessreviews",
+	"authorization.k8s.io/v1beta1/selfsubjectrulesreviews",
+	"authorization.k8s.io/v1beta1/subjectaccessreviews",
+	"extensions/v1beta1/replicationcontrollers",
+)
+
+// GVRToStorageVersionHash shouldn't change unless we intentionally change the
+// storage version of a resource.
+var GVRToStorageVersionHash = map[string]string{
+	"v1/configmaps":             "O/wmlZTvZJIo6adLqwDwQu/JHVrMb77jGjgugNQjiP4=",
+	"v1/endpoints":              "O/wmlZTvZJIo6adLqwDwQu/JHVrMb77jGjgugNQjiP4=",
+	"v1/events":                 "O/wmlZTvZJIo6adLqwDwQu/JHVrMb77jGjgugNQjiP4=",
+	"v1/limitranges":            "O/wmlZTvZJIo6adLqwDwQu/JHVrMb77jGjgugNQjiP4=",
+	"v1/namespaces":             "O/wmlZTvZJIo6adLqwDwQu/JHVrMb77jGjgugNQjiP4=",
```

Comment:
Something is wrong, these should all be different. (Another benefit of checking them in!)

Reply:
Looks good except for this. Please squash once you fix this and then it should be good to go.

Reply:
this is just the stored group/version, not the full storage path, so all the things storing as group:"",version:"v1" should match, right?

Reply:
Right. This was expected by the design in the [KEP](https://github.com/kubernetes/enhancements/blob/master/keps/sig-api-machinery/35-storage-version-hash.md).

The hash for the storage path is `ResourceID` and is captured in a different [KEP](https://github.com/kubernetes/enhancements/blob/master/keps/sig-api-machinery/34-storage-hash.md).

Reply:
The first KEP does not actually list what will go into the hash. Sorry, I should have noticed that when reading the KEP, that's embarrassing. I expected it ("storage version") to be a GVK, not just a GV--the things we actually write are always encoded as some GVK...

My intuition is that we want each hash to appear in the smallest number of places possible; it feels safer, like it might be easier in the future (considering client usage) to change from hashing GVK to GV than it would to change from hashing GV to GVK.

@deads2k were you expecting this to be the same across multiple unrelated kinds?

Reply:
I think the stored kind should (and is?) always match the [kind](https://github.com/kubernetes/kubernetes/blob/493b26105789bf88575b78cce9e8474e20ae3f48/staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go#L940) listed in the discovery doc, the only gap between what's stored in etcd and what's shown in the discovery doc is the encoding group and version, so hashing the group+version is sufficient.

Reply:
I had thought it was just a groupversion, so all the kinds sharing one would have the same hash.  I have nothing against using the GVK instead.  Doing so would allow us to move a kind inside of a groupversion and I can't think of a downside.